compiler/circle-mpqsolver/src/bisection/BisectionSolver.cpp

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "BisectionSolver.h"
+#include "DepthParameterizer.h"
+#include "ErrorMetric.h"
+
+#include <luci/ImporterEx.h>
+#include <luci/Log.h>
+
+#include <cmath>
+#include <iostream>
+
+using namespace mpqsolver::bisection;
+
+namespace
+{
+
+class BisectionOptionsImpl final : public BisectionSolver::Options
+{
+public:
+  virtual void enable(Q16AtInput) final;
+  virtual bool query(Q16AtInput) final;
+
+private:
+  Q16AtInput _q16AtInput = Q16AtInput::True;
+};
+
+void BisectionOptionsImpl::enable(Q16AtInput q16AtInput) { _q16AtInput = q16AtInput; }
+bool BisectionOptionsImpl::query(Q16AtInput q16AtInput) { return _q16AtInput == q16AtInput; }
+
+std::unique_ptr<luci::Module> read_module(const std::string &path)
+{
+  luci::ImporterEx importerex;
+  auto module = importerex.importVerifyModule(path);
+  if (module.get() == nullptr)
+  {
+    std::cerr << "Failed to load " << path << std::endl;
+    return nullptr;
+  }
+
+  return module;
+}
+
+} // namespace
+
+BisectionSolver::BisectionSolver(const std::string &input_data_path, float qerror_ratio)
+  : _input_data_path(input_data_path), _qerror_ratio(qerror_ratio), _qerror(0.f),
+    _quantizer(std::make_unique<Quantizer>("uint8", "uint8"))
```

Comment:
input/output_dtype is fixed?

Reply:
Currently it is, but we may introduce input/output dtype as parameters for the program, and make "uint8" default.